crates/env/src/topics.rs

```diff
@@ -0,0 +1,204 @@
+// Copyright 2018-2020 Parity Technologies (UK) Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Docs
+
+use crate::EnvTypes;
+
+/// The concrete implementation that is guided by the topics builder.
+///
+/// To be implemented by the on-chain and off-chain environments respectively.
+#[doc(hidden)]
```

Comment:
True but I think marking each and every component that shall be hidden within such a small module increases readability since it restates the intent of the hidden entities locally. For example the `Topics` trait is not hidden and maybe some others will follow. If they were not hidden some developer in the future might think - "oh damn, we forgot to re-export those to the crate level, let's do that!". And that would be bad.

Reply:
Makes sense.